Source/WebKit/UIProcess/API/wpe/WPEWebView.h

```diff
@@ -115,27 +130,40 @@ class View : public API::ObjectImpl<API::Object::Type::View> {
     static WebKit::WebPageProxy* platformWebPageProxyForGamepadInput();
 #endif
 
+    void updateAcceleratedSurface(uint64_t);
+
+    void setCursor(const WebCore::Cursor&);
+
 private:
-    View(struct wpe_view_backend*, const API::PageConfiguration&);
+    View(struct wpe_view_backend*, WPEDisplay*, const API::PageConfiguration&);
 
     void setSize(const WebCore::IntSize&);
     void setViewState(OptionSet<WebCore::ActivityState>);
     void handleKeyboardEvent(struct wpe_input_keyboard_event*);
 
+#if ENABLE(TOUCH_EVENTS)
+    Vector<WebKit::WebPlatformTouchPoint> touchPointsForEvent(WPEEvent*);
+#endif
+
     std::unique_ptr<API::ViewClient> m_client;
 
 #if ENABLE(TOUCH_EVENTS)
     std::unique_ptr<WebKit::TouchGestureController> m_touchGestureController;
+    HashMap<uint32_t, GRefPtr<WPEEvent>> m_touchEvents;
 #endif
     std::unique_ptr<WebKit::PageClientImpl> m_pageClient;
     RefPtr<WebKit::WebPageProxy> m_pageProxy;
     WebCore::IntSize m_size;
     OptionSet<WebCore::ActivityState> m_viewStateFlags;
+    std::unique_ptr<WebKit::AcceleratedBackingStoreDMABuf> m_backingStore;
 
     struct wpe_view_backend* m_backend;
+    WPEDisplay* m_wpeDisplay { nullptr };
```

Comment:
The display is owned by the WPEWebView owners (WebKitWebView for glib API, PlatformWebViewClientWPE for C API). WPEWebView is always created with unique_ptr so it's always deleted before its owner. I don't think it makes sense to use a weak reference, because it can't be deleted while the WPEWebView is alive, but I don't mind to use GRefPtr, because even though it's useless, there's no risk of cycles.

Reply:
hmm, but we don't really need to keep the display, since we can get if from the view, so I'm going to remove it.